libmamba/src/core/solver.cpp

```diff
@@ -227,6 +227,15 @@ namespace mamba
         if (!chan)
             return false;
 
+        // Check the subdirectory.
+        // This assumes, that explicitly specified subdirectories
+        // (e.g. conda-forge/linux-64::mamba) result in the platform member of needle
+        // only containing one entry. When the subdirectory is omitted, the paltfom
+        // member instead contains implicit <platform> and noarch entries.
```

Comment:
Typo in "platform", also I find this a little easier to parse:

```suggestion
        // This assumes that explicitly specified subdirectories
        // (e.g. conda-forge/linux-64::mamba) result in needle.platforms.size() == 1 (e.g. needle.platforms() == {"linux-64"}). If the subdirectory is omitted from the spec, needle.platforms() is {"noarch", "<platform>"}.
```

Reply:
I also like this! Not sure how useful the comment is in general, or if that is "common knowledge". I just wanted to make the `== 1` comparison come a little less out of the blue.